src/main/java/algorithms/curated170/medium/threesumsmaller/ThreeSumSmallerRecursion.java

```diff
@@ -0,0 +1,51 @@
+package algorithms.curated170.medium.threesumsmaller;
+
+import java.util.Arrays;
+
+public class ThreeSumSmallerRecursion {
+
+    int[] nums;
+    int target;
+    int rightBoundary;
+
+    public int threeSumSmaller(int[] nums, int target) {
+        if (nums.length < 3) {
+            return 0;
+        }
+        
+        this.nums = nums;
+        this.target = target;
+        rightBoundary = nums.length - 1;
+        Arrays.sort(nums);
+
+        return reduceTwoSumsSmaller(0);
+    }
+
+    public int reduceTwoSumsSmaller(int start) {
+        int currTarget = target - nums[start];
+        int p1 = start + 1, p2 = rightBoundary;
+        int count = 0;
+        boolean foundBounds = false;
+        while (p1 < p2) {
+            if (nums[p1] + nums[p2] >= currTarget) {
+                if (!foundBounds) {
+                    rightBoundary--;
+                }
+                p2--;
+            } else {
+                foundBounds = true;
+                count += p2 - p1;
+                p1++;
+            }
+        }
+        return count == 0 ? 0 : count + reduceTwoSumsSmaller(start + 1);
+    }
```

Comment:
A little bit of simplification and readability fix.

```suggestion
    public int reduceTwoSumsSmaller(int start) {
        int currTarget = target - nums[start];
        int left = start+1, right = rightBoundary;
        int count = 0;
        while (left < right) {
            if (nums[left] + nums[right] < currTarget) {
                  count += right - left;
                  left++;
            } else {
                  right--;
               
            }
        }
        return count == 0 ? 0 : count + reduceTwoSumsSmaller(start + 1);
    }
```

Reply:
How do we make use of the right boundary in this? I don't think itmis being used.

Reply:
We should keep the initial state of `rightBoundary ` for each recursion. If we decrement it (instead of `right`), it will not work.